### PR TITLE
Fix voting a player to pause doesn't work when sv_pauseable is on.

### PR DESCRIPTION
**Pull request by d3fault**
When sv_pauseable is on, force_pause is used which doesn't appear
to work. When sv_pauseable is off, set_team is used, which does
work.

Special thanks to SP | Someone for reporting and testing.

### CODE REVIEW

File: src/game/server/gamecontext.cpp [hunk 1]
```diff
@@ -1447,18 +1447,11 @@ void CGameContext::OnMessage(int MsgID, CUnpacker *pUnpacker, int ClientID)
...
 					return;
 				}
-
-				if(g_Config.m_SvPauseable && g_Config.m_SvVotePause)
-				{
-					str_format(aChatmsg, sizeof(aChatmsg), "'%s' called for vote to pause '%s' for %d seconds (%s)", Server()->ClientName(ClientID), Server()->ClientName(SpectateID), g_Config.m_SvVotePauseTime, aReason);
-					str_format(aDesc, sizeof(aDesc), "Pause '%s' (%ds)", Server()->ClientName(SpectateID), g_Config.m_SvVotePauseTime);
-					str_format(aCmd, sizeof(aCmd), "force_pause %d %d", SpectateID, g_Config.m_SvVotePauseTime);
-				}
 				else
```
Comment (heinrich5991) [inline src/game/server/gamecontext.cpp]: This `else` is bogus.

**Comment by d3fault** [comment]
@heinrich5991 force_pause does the same thing as set_team, so why should I waste my time fixing broken duplicate code? I'm also too [lazy|efficient] to go and remove every use of force_pause; that can/should go in it's own "cleanup" commit, since it won't technically change anything at all functionality-wise.

and yea the catch-all else did look weird to me too, but it's not _my_ else

**Comment by heinrich5991** [comment]
`set_team` kills the tee, `force_pause` only pauses it. This is the difference between clicking "Spectate" in the menus (which kills you) and writing `/spec` in chat (which just removes your tee until you write `/spec` again).

>and yea the catch-all else did look weird to me too, but it's not my else

The lone `else` block is there because you removed the corresponding `if`.

**Comment by d3fault** [comment]
Wops yea for some reason I thought that else was following a chain of "else ifs". Thanks, fix'd.

I still think this patch is fine. It doesn't change the effect of writing /spec (that doesn't now kill you). Why do you say it's ok for "vote to spec" to kill the tee when sv_pauseable is off but it's suddenly not ok to kill the tee when sv_pauseable is on?

**Comment by heinrich5991** [comment]
It's not ok to kill the tee because there is a better option available - force-`/spec` it. Why would you ruin someone's race just because they might be afk-blocking somewhere on the map, if there's another option.

**Comment by d3fault** [comment]
Killing the tee is already the current functionality, so you're proposing to change the game whereas I'm fixing a case where it's broken.

I still don't see why sv_pauseable is the deciding factor for whether or not to kill them. Isn't sv_pauseable meant to decide only whether or not the tee goes invisible when the player types /spec? Or are you using 1 variable for multiple purposes?

I'd be fine with "vote to spec" merely pausing the player for a few seconds instead of killing them (note: I've never seen this happen. maybe it's because I only DDRace?), but that should be the result of the vote regardless of the value of sv_pauseable. I confirmed that force_pause doesn't even work, and since I'm fine with killing (and it already happens!) I'm not about to go hunt down+fix force_pause. Only once force_pause works should it be used.

**Comment by heinrich5991** [comment]
`sv_pauseable` determines whether the mapper allows the usage of `/spec` that removes your tee from the map until you un`/spec`. This is determined per map because it potentially allows you to cheat – but since `/spec` is allowed for everyone, that's fine. So if we're allowed to `/spec` (`sv_pauseable 1`), that's what the vote to spec should (and used to) do.

If we're not allowed to `/spec` by the map, then that shouldn't be overriden by the vote; otherwise people could cheat maps by voting to spec instead of using `/spec`.

**Comment by Learath2** [comment]
If you can't be bothered to fix force_pause then you can't fix the issue. The intended beahavior of the original code is clear and a change from that isn't desirable per what @heinrich5991 said.

**Pull request closed without merging**